pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TransactionBufferHandlerImpl.java

```diff
@@ -336,8 +291,42 @@ protected OpRequestSend newObject(Handle<OpRequestSend> handle) {
         };
     }
 
+    public CompletableFuture<ClientCnx> getClientCnxWithLookup(String topic) {
+        return pulsarClient.getConnection(topic);
+    }
+
     public CompletableFuture<ClientCnx> getClientCnx(String topic) {
-        return ((PulsarClientImpl) pulsarClient).getConnection(topic);
+        NamespaceService namespaceService = pulsarService.getNamespaceService();
+        CompletableFuture<NamespaceBundle> nsBundle = namespaceService.getBundleAsync(TopicName.get(topic));
```

Comment:
may we should think about namespace not exist

Reply:
The `getBrokerServiceUrlAsync ` will wrap this exception to CompletableFuture

Reply:
Process the BrokerMetadataException and add  `testDeleteNamespaceBeforeCommit` for this .

Reply:
Process the BrokerMetadataException and add  `testDeleteNamespaceBeforeCommit` for this .